oracle/stork_fetcher.go

```diff
@@ -167,7 +168,15 @@ func (f *storkFetcher) startReadingMessages() error {
 			// Update the cached asset pairs
 			newPairs := make(map[string]*oracletypes.AssetPair, len(assetIds))
 			for _, assetId := range assetIds {
-				pair := ConvertDataToAssetPair(data[assetId], assetId)
+				asset := data[assetId]
+				if len(asset.SignedPrices) == 0 {
+					log.Warningln("no signed prices found for asset:", assetId)
+					continue
+				}
+
+				tsReference := asset.SignedPrices[0].TimestampedSignature.Timestamp
```

Comment:
Maybe do the conversion once here
```suggestion
				tsReferenceInSeconds := ConvertTimestampToSecond(asset.SignedPrices[0].TimestampedSignature.Timestamp)
```